test/e2e/util/command.go

```diff
@@ -61,30 +71,37 @@ func (b *cmdBuilder) Exec() (string, string, error) {
 	return stdout.String(), stderr.String(), err
 }
 
-func (b *cmdBuilder) Start() (io.Reader, io.Reader, func() error, <-chan error) {
+func (b *cmdBuilder) Start(t *testing.T, terminateTimeout time.Duration) (io.Reader, io.Reader, func()) {
 	stdout := newSyncBuffer()
 	stderr := newSyncBuffer()
 	b.cmd.Stdout = io.MultiWriter(os.Stdout, stdout) // we want to see full `getenvoy` output in the test log
 	b.cmd.Stderr = io.MultiWriter(os.Stderr, stderr)
-	errs := make(chan error, 1)
-	if err := b.cmd.Start(); err != nil {
-		errs <- err
-		close(errs)
-		return stdout, stderr, func() error { return nil }, errs
-	}
+	err := b.cmd.Start()
+	require.NoError(t, err, `error starting [%v]`, b)
+
+	errc := make(chan error, 1)
 	go func() {
-		defer close(errs)
-		if err := b.cmd.Wait(); err != nil {
-			errs <- err
-		}
+		errc <- b.cmd.Wait()
 	}()
-	canceled := false
-	cancel := func() error {
-		if canceled {
-			return nil
+
+	// Ensure terminate is only called once. As this happens from a single thread, there's no need to lock.
```

Comment:
maybe we could use `sync.Once` rather than using closure capturing the state `terminated` :smile:

Reply:
ah - this would make the consistency of the returned function of these utilities. Fine as it is!